lib/node_modules/@stdlib/repl/lib/editor_actions.js

```diff
@@ -0,0 +1,577 @@
+		'sequence': '\u001F'
+	});
+});
+
+/**
+* Yank-pop the next string from the redo stack.
+*
+* @name redo
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'redo', function redo() {
+	this._ttyWrite.call( this._rli, null, {
+		'sequence': '\u001E'
+	});
+});
+
+/**
+* Transposes the characters about cursor.
+*
+* @name transposeAboutCursor
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'transposeAboutCursor', function transposeAboutCursor() {
+	var c = this._rli.line[ this._rli.cursor - 1 ];
+	this.deleteLeft();
+	this.moveRight();
+	this._rli.write( c );
+});
+
+/**
+* Changes the next word to uppercase.
+*
+* @name uppercaseNextWord
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'uppercaseNextWord', function uppercaseNextWord() {
+	this._modifyNextWord( uppercase );
+});
+
+/**
+* Changes the next word to titlecase.
+*
+* @name capitalizeNextWord
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'capitalizeNextWord', function capitalizeNextWord() {
+	this._modifyNextWord( capitalize );
+});
+
+/**
+* Changes the next word to lowercase.
+*
+* @name lowercaseNextWord
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'lowercaseNextWord', function lowercaseNextWord() {
+	this._modifyNextWord( lowercase );
+});
+
+/**
+* Clears the entire REPL screen and scrollback history.
+*
+* @name clearScreen
+* @memberof EditorActions.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'clearScreen', function clearScreen() {
+	this._ttyWrite.call( this._rli, null, {
+		'name': 'l',
+		'ctrl': true
+	});
+});
+
+/**
+* Callback which should be invoked **before** a "keypress" event is processed by a readline interface.
+*
+* @name beforeKeypress
+* @memberof EditorActions.prototype
+* @type {Function}
+* @param {string} data - input data
+* @param {(Object|void)} key - key object
+* @returns {boolean} boolean indicating whether an editor action was triggered
+*/
+setNonEnumerableReadOnly( EditorActions.prototype, 'beforeKeypress', function beforeKeypress( data, key ) {
+	var possibleKeys;
+	var actionKeys;
+	var action;
+	var i;
+	var j;
+	var k;
+
+	if ( !key ) {
+		return false;
+	}
+	possibleKeys = parseKey( key );
+	for ( i = 0; i < EDITOR_ACTIONS.length; i++ ) {
+		actionKeys = this._keybindings[ EDITOR_ACTIONS[ i ] ];
+		action = this._actions[ EDITOR_ACTIONS[ i ] ];
+		if ( !actionKeys ) {
+			continue;
+		}
+		for ( j = 0; j < actionKeys.length; j++ ) {
+			for ( k = 0; k < possibleKeys.length; k++ ) {
+				if ( deepEqual( possibleKeys[ k ], actionKeys[ j ] ) ) {
```

Comment:
Deep equality could be an expensive check. Is there a potential lighter weight means?

Reply:
Given that we know the keys, can we not write a small custom function which explicitly compares known keys?

Reply:
For two keys to be equal, their four properties should be equal (`name`, `ctrl`, `meta`, `shift`). Wouldn't deep equality do just that?

Reply:
Yes, but it is the difference between comparing values having an unknown schema and those having a known schema. For the former, you need to dynamically resolve the list of keys and then iterate. For the latter, you can skip resolution and directly compare without iteration. It is something of a micro-optimization, but my main concern is preventing lag, especially as we continue to add various keystroke listeners.

Reply:
Makes sense. I'll make a helper function, or better, a module like `containsKeys` which will take an array of possible keybindings (as one action has multiple keybindings set) and the user input key.

Reply:
Seems reasonable.